### PR TITLE
Implement layout component

### PR DESCRIPTION
**Pull request by ajronn**

### CODE REVIEW

File: src/ui/index.ts [hunk 1]
```diff
@@ -23,3 +23,4 @@ export * from './tag';
 export * from './tags';
 export * from './textarea-field';
 export * from './tree';
+export * from './layout';
```
Comment (polubis) [inline src/ui/index.ts:26]: Alph order

File: src/ui/layout/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React, { ReactNode } from "react"
+
+import csx from "./Layout.scss"
+
+namespace Layout {
+    export interface Props {
+        className?: string;
+        children: ReactNode;
+        scroll?: boolean;
```
Comment (polubis) [inline src/ui/layout/Layout.tsx:9]: scroll is a little bit confusing. This props adds additional height - maybe overgrown props will be better ?

File: src/ui/layout/tests/Layout.test.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import { Layout } from "..";
+
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:5]: Format all files

File: src/ui/layout/tests/Layout.test.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+describe('<Layout>', () => {
+    it("shows children", () => {
+        const child = <div data-testid="child">Test</div>;
+        render(<Layout>{child}</Layout>)
+        expect(screen.getByTestId("child")).toBeDefined();
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:11]: toBeInTheDocument()

File: src/ui/layout/tests/Layout.test.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import { Layout } from "..";
+
+
+describe('<Layout>', () => {
+    it("shows children", () => {
+        const child = <div data-testid="child">Test</div>;
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:9]: Don't user data-testid without a real reason. Replace this with fe. <input placeholder="SomeTitle" /> and use screen.getByPlaceholderText to search.

Also remove child variable - there is one good rule for creating variables. 
Create only when:
- reduces duplication,
- makes code cleaner

File: src/ui/layout/tests/Layout.test.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        render(<Layout>{child}</Layout>)
+        expect(screen.getByTestId("child")).toBeDefined();
+    })
+
+    it("style pass correctly", () => {
+        const child = <div data-testid="child">Test</div>;
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:15]: remove variable and test id.
Change test title - applies className - without correctly

File: src/ui/layout/tests/Layout.test.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        const { container } = render(<Layout className={"foo"}>{child}</Layout>)
+        expect(container.firstChild).toHaveClass('foo');
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:17]: container.querySelector('.yourclass').toBeInTheDocument()

File: src/ui/layout/tests/Layout.test.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    })
+
+    it("scroll var change height property", () => {
+        const child = <div data-testid="child">Test</div>;
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:21]: If test is more complicated - add enters between ARANGE, ACT, ASSERT steps.

Also test should be named - displays browser scroll for overgrown property

File: src/ui/layout/tests/Layout.test.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        render(<Layout scroll={true}>{child}</Layout>)
+        const props = {
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:23]: Remove not needed variables

File: src/ui/layout/tests/Layout.test.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+            scroll: true,
+            children: child
+        }
+        const layoutClass = Layout(props).type.styledComponentId;
+        const layoutRoots = document.getElementsByClassName(layoutClass);
+        const style = window.getComputedStyle(layoutRoots[0]);
+        expect(style.height).toBe('110%');
```
Comment (polubis) [inline src/ui/layout/tests/Layout.test.tsx:30]: Implementation details test - you should check if browser have scroll displayed and you should use Stubs for testing.
Try to display this component for different situations via Stub. 

For example:

- for UI which is small,
- for UI which have too many elements and scroll is displayed by default,
- for elements that are on the verge of the appearance of a scroll

**Review by polubis — CHANGES_REQUESTED** [review]

File: src/ui/index.ts (hunk 1 above)
Comment (ajronn) [reply src/ui/index.ts:26]: I think this is not alph order... 
image

File: src/ui/layout/tests/Layout.test.tsx (hunk 6 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:15]: done

File: src/ui/layout/tests/Layout.test.tsx (hunk 7 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:17]: done

File: src/ui/layout/tests/Layout.test.tsx (hunk 9 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:23]: done

File: src/ui/layout/Layout.tsx (hunk 2 above)
Comment (ajronn) [reply src/ui/layout/Layout.tsx:9]: done

File: src/ui/layout/tests/Layout.test.tsx (hunk 3 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:5]: done

File: src/ui/layout/tests/Layout.test.tsx (hunk 4 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:11]: done

File: src/ui/index.ts (hunk 1 above)
Comment (polubis) [reply src/ui/index.ts:26]: Should be aplha

File: src/ui/layout/tests/Layout.test.tsx (hunk 5 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:9]: done

File: src/ui/layout/tests/Layout.test.tsx (hunk 8 above)
Comment (ajronn) [reply src/ui/layout/tests/Layout.test.tsx:21]: done

File: src/ui/layout/Layout.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+        className?: string;
+        children: ReactNode;
+        overgrown?: boolean;
+    }
+}
+
+const Layout = ({ className, children, overgrown = false }: Layout.Props) => {
+    return (
+        <div className={`${csx.layout} ${className}`}
+            style={{ height: overgrown ? "110%" : "100%" }}>
```
Comment (Czaroos) [inline src/ui/layout/Layout.tsx:16]: ```<div className={`${csx.layout} ${overgrown ? csx.overgrown : ''} ${className}`}```

In scss file:
```
.overgrown {
    height: 110%;
}
```

Then remove inline styling

**Review by Czaroos — CHANGES_REQUESTED** [review]

**Review by polubis — APPROVED** [review]

**Pull request merged**